hypha/apply/users/models.py

```diff
@@ -87,18 +93,49 @@ def create_superuser(self, email, password, **extra_fields):
 
         return self._create_user(email, password, **extra_fields)
 
+    def _extract_model_params(self, defaults, **kwargs):
+        """
+        Prepare `params` for creating a model instance based on the given kwargs;
+        A copied method from model's query.py; for use by get_or_create_and_notify().
+        """
+        defaults = defaults or {}
+        params = {k: v for k, v in kwargs.items() if LOOKUP_SEP not in k}
+        params.update(defaults)
+        property_names = self.model._meta._property_names
+        invalid_params = []
+        for param in params:
+            try:
+                self.model._meta.get_field(param)
+            except exceptions.FieldDoesNotExist:
+                # It's okay to use a model's property if it has a setter.
+                if not (param in property_names and getattr(self.model, param).fset):
+                    invalid_params.append(param)
+        if invalid_params:
+            raise exceptions.FieldError(
+                "Invalid field name(s) for model %s: '%s'." % (
+                    self.model._meta.object_name,
+                    "', '".join(sorted(invalid_params)),
+                ))
+        return params
+
     def get_or_create_and_notify(self, defaults=dict(), site=None, **kwargs):
         # Set a temp password so users can access the password reset function if needed.
         temp_pass = BaseUserManager().make_random_password(length=32)
         temp_pass_hash = make_password(temp_pass)
         defaults.update(password=temp_pass_hash)
-        user, created = self.get_or_create(defaults=defaults, **kwargs)
-        if created:
+        user = self.filter(email__iexact=kwargs.get('email')).first()  # case insensitive matching
```

Comment:
use `get_user_by_email` here?

Reply:
At the time of user creation, we only need a case-insensitive search.  We can also use `get_user_by_email` but for that, we will need to pass an extra argument and will need to update the logic of the function a bit.  But yes, our logic will only be in one place. Let me try with that logic and see if it looks good.

Reply:
Let's not do this